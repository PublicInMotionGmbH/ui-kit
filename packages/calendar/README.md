# Talixo Calendar

UI Component which represents Calendar

## How to install

Package is available as `@talixo/calendar` in NPM registry, so you can use it in your project
using `npm install @talixo/calendar --save` or `yarn add @talixo/calendar`.

## Requirements

Your package should additionally have some extra dependencies:

- `prop-types: ^15.6.1`
- `react: ^16.2.0`
- `react-dom: ^16.2.0`
- `moment: ^2.22.0`
- `react-dates: ^16.5.0`
- `@talixo/shared: ^0.1.0`

These packages are required by `@talixo/calendar`, but you have to install them manually,
to avoid having different versions of these in your application.

## Supported props

It allows any props which are allowed for `div`. Additionally, it handles some differently:

Property name      | Type                  | Default         | Description
-------------------|-----------------------|:---------------:|--------------------------------
className          | string                | n/a             | additional class name passed to wrapper
displayFormat      | string                | `'D MMM YYYY'`  | date display format
dayAriaLabelFormat | string                | n/a             | day aria label format
monthFormat        | string                | n/a             | month format
phrases            | object                | n/a             | phrases for i18n
weekDayFormat      | string                | n/a             | week day format
placeholder        | string                | n/a             | placeholder text
value              | string or Moment date | self-controlled | current date
onChange           | function              | n/a             | event handler when date is changed
onFocus            | function              | n/a             | event handler when calendar is focused
onBlur             | function              | n/a             | event handler when calendar lost focus

## Changelog

- **0.1.0** - initial version
