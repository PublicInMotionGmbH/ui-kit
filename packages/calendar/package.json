{
  "name": "@talixo/calendar",
  "version": "1.0.0-alpha.16",
  "description": "UI Component which represents Calendar",
  "main": "dist/index.js",
  "module": "index.js",
  "jsnext:main": "index.js",
  "scripts": {
    "test": "node ../../scripts/runTests.js --only calendar",
    "prepublishOnly": "npm run build",
    "build": "node ../../scripts/buildPackages.js --only calendar"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "Public in Motion GmbH",
  "repository": {
    "type": "git",
    "url": "https://github.com/PublicInMotionGmbH/ui-kit.git"
  },
  "bugs": {
    "url": "https://github.com/PublicInMotionGmbH/ui-kit/issues"
  },
  "license": "MIT",
  "devDependencies": {
    "@talixo/icon": "^1.0.0-alpha.16",
    "@talixo/shared": "^1.0.0-alpha.16",
    "moment": "^2.22.1",
    "prop-types": "^15.6.1",
    "react": "^16.2.0",
    "react-dates": "^16.5.0",
    "react-dom": "^16.2.0"
  },
  "peerDependencies": {
    "@talixo/icon": "^1.0.0-alpha.16",
    "@talixo/shared": "^1.0.0-alpha.16",
    "moment": "^2.22.1",
    "prop-types": "^15.6.1",
    "react": "^16.2.0",
    "react-dates": "^16.5.0",
    "react-dom": "^16.2.0"
  }
}
