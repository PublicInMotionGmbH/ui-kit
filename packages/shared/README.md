# Talixo Shared

Shared data between whole UI Kit.

## How to install

Package is available as `@talixo/shared` in NPM registry, so you can use it in your project
using `npm install @talixo/shared --save` or `yarn add @talixo/shared`.

## Requirements

Your package should additionally have some extra dependencies:

- `classnames: ^2.2.5`

These packages are required by `@talixo/shared`, but you have to install them manually,
to avoid having different versions of these in your application.
