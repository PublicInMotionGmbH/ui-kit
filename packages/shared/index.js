import applyAnyClassNameModifiers from './src/applyAnyClassNameModifiers'
import applyClassNameModifiers from './src/applyClassNameModifiers'
import buildClassName from './src/buildClassName'
import prefix from './src/prefix'

export {
  applyAnyClassNameModifiers,
  applyClassNameModifiers,
  buildClassName,
  prefix
}
