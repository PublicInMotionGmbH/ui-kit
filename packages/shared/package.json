{
  "name": "@talixo/shared",
  "version": "1.0.0",
  "description": "Talixo common configuration",
  "author": "Public in Motion GmbH",
  "main": "index.js",
  "module": "index.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/PublicInMotionGmbH/ui-kit.git"
  },
  "bugs": {
    "url": "https://github.com/PublicInMotionGmbH/ui-kit/issues"
  },
  "devDependencies": {
    "rollup": "^0.57.1"
  },
  "dependencies": {
    "normalize.css": "^8.0.0"
  }
}
