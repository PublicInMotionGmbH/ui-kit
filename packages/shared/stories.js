import React from 'react'
import { createStoriesFactory } from './story'

// Load first paragraph from README file
const readme = ''

// Create factories for story
const addStory = createStoriesFactory('Typography and basics', module)

// Stories

addStory('Headers', readme, () => (
  <div>
    <h1>h1. Header</h1>
    <h2>h2. Header</h2>
    <h3>h3. Header</h3>
    <h4>h4. Header</h4>
    <h5>h5. Header</h5>
    <h6>h6. Header</h6>

    <h1>h1. Header <small>Secondary text</small></h1>
    <h2>h2. Header <small>Secondary text</small></h2>
    <h3>h3. Header <small>Secondary text</small></h3>
    <h4>h4. Header <small>Secondary text</small></h4>
    <h5>h5. Header <small>Secondary text</small></h5>
    <h6>h6. Header <small>Secondary text</small></h6>
  </div>
))

addStory('Text styling', readme, () => (
  <div>
    <h2>Paragraphs</h2>

    <p>
      Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis.
      Maecenas malesuada elit lectus felis, malesuada ultricies.
      Curabitur et ligula. Ut molestie a, ultricies porta urna.
      Vestibulum commodo volutpat a, convallis ac, laoreet enim. Phasellus fermentum in, dolor.
    </p>

    <p>
      Pellentesque facilisis. Nulla imperdiet sit amet magna.
      Vestibulum dapibus, mauris nec malesuada fames ac turpis velit, rhoncus eu, luctus et interdum adipiscing wisi.
    </p>

    <p>
      Aliquam erat ac ipsum. Integer aliquam purus.
      Quisque lorem tortor fringilla sed, vestibulum id, eleifend justo vel bibendum sapien massa ac turpis
      faucibus orci luctus non, consectetuer lobortis quis, varius in, purus.
      Integer ultrices posuere cubilia Curae, Nulla ipsum dolor lacus, suscipit adipiscing.
      Cum sociis natoque penatibus et ultrices volutpat.
    </p>

    <h2>Inline styling</h2>

    <p>
      We may have some <strong>strong elements</strong> inside our text, as well as some <em>emphasises</em>.
      When we want to calculate 2<sup>3</sup> + x<sub>1</sub>, we may have problems, but it's easy to show it.
      <br />
      In some circumstances, mostly describing our API or other stuff we may also try to show some <code>short code snippet</code>.
      It happens that we may also use <kbd>Ctrl+C</kbd> to hint user with keyboard shortcuts.
    </p>

    <p>
      It may happen that we would like to <mark>highlight</mark> some content inside our dashboards, in case of some search engine.
    </p>

    <p>
      <del>We do not make any mistakes.</del> When we do, we can write it semantically marking that anyway.
    </p>

    <p>
      When we use some shortcuts like <abbr title='Looks good to me'>LGTM</abbr>, we can show such information for user.
    </p>

    <h2>Quotes and testimonials</h2>

    <blockquote>
      This was one of the best experiences I have ever had with a cab company.
      I had problems at the airport and the driver stayed with me for over an hour and helped me sort everything out.
      I would recommend this company to anyone. Thank you for such fabulous service!
    </blockquote>

    <blockquote>
      This was one of the best experiences I have ever had with a cab company.
      I had problems at the airport and the driver stayed with me for over an hour and helped me sort everything out.
      I would recommend this company to anyone. Thank you for such fabulous service!

      <footer>
        <cite>R.M.</cite>
      </footer>
    </blockquote>
  </div>
))

addStory('RTL: Headers', readme, () => (
  <div dir='rtl'>
    <h1>h1. כּוֹתֶרֶת</h1>
    <h2>h2. כּוֹתֶרֶת</h2>
    <h3>h3. כּוֹתֶרֶת</h3>
    <h4>h4. כּוֹתֶרֶת</h4>
    <h5>h5. כּוֹתֶרֶת</h5>
    <h6>h6. כּוֹתֶרֶת</h6>

    <h1>h1. כּוֹתֶרֶת <small>טקסט משני</small></h1>
    <h2>h2. כּוֹתֶרֶת <small>טקסט משני</small></h2>
    <h3>h3. כּוֹתֶרֶת <small>טקסט משני</small></h3>
    <h4>h4. כּוֹתֶרֶת <small>טקסט משני</small></h4>
    <h5>h5. כּוֹתֶרֶת <small>טקסט משני</small></h5>
    <h6>h6. כּוֹתֶרֶת <small>טקסט משני</small></h6>
  </div>
))

addStory('RTL: Text styling', readme, () => (
  <div dir='rtl'>
    <h2>פסקאות</h2>

    <p>
      אחד אל בדפים חרטומים, או פיסול צרפתית ויקיפדיה היא, אם מדריכים הגולשות כתב. ב עוד המחשב ברוכים עקרונות. ספינות מדריכים אקטואליה אחר על.
      זאת אם ביולוגיה אדריכלות, עוד ראשי וקשקש מתמטיקה ב. ויש בה רביעי תרומה לציין. זקוק המדינה גם עוד.
      זאת דת לתרום הבאים נוסחאות, אם פיסיקה סטטיסטיקה אווירונאוטיקה רבה.
    </p>

    <p>
      נפלו למנוע מונחונים או כדי. אחרות תקשורת בה אחר. מיזם יסוד רביעי בה מלא, הבקשה בדפים על זכר. כלל או העזרה ממונרכיה, כלל גם הטבע אנציקלופדיה, אנא אל שמות יסוד.
    </p>

    <p>
תיקונים והגולשים אל בדף, של בדף בישול יוצרים העברית. עזה אם והוא הארץ בידור, תחבורה משופרות עזה אם.
      מה רוסית בארגז הגולשות צ'ט, קבלו מוסיקה ב אחד, מדעי אדריכלות בדף בה. מדע שתפו ויקימדיה אם.
      אדריכלות אווירונאוטיקה אתה של, תרבות ולחבר צ'ט דת.
      לחשבון לויקיפדיה עזה אל, תנך דרכה בכפוף מיוחדים של.
      שתי מה רפואה אודות, כיצד רקטות סטטיסטיקה דת שמו. עוד אקראי וקשקש מיוחדים אם,
      לוח גם העזרה בהתייחסות. מה מונחים אספרנטו קרן. אינו לעריכה של חפש, בה מאמר מוגש אנא.
      בקר הרוח מתמטיקה את, בה החלה יסוד הגולשות קרן. בה לימודים ממונרכיה ואלקטרוניקה עזה,
      חופשית מועמדים מדע של. מדע עזרה אחרות המלצת של, דת יוני אנגלית כתב.
    </p>

    <h2>עיצוב מוטבע</h2>

    <p>
      אולי יש לנו<strong> כמה אלמנטים חזקים בתוך הטקסט </strong>שלנו, כמו גם כמה מדגיש.
      ב במקרים מסוימים, בעיקר לתאר את ה- API שלנו או דברים אחרים אנו עשויים<code> גם לנסות להציג קטע קוד קצר. זה קורה</code>כי אנו יכולים גם >להשתמ
      <kbd>Ctrl+C</kbd> לרמז למשתמש עם קיצורי מקשים.
    </p>

    <p>
      זה יכול לקרות כי אנחנו רוצים להדגיש קצת תוכן בתוך לוחות.<mark> המחוונים שלנו, במקרה של מנוע </mark>החיפוש
    </p>

    <p>
      <del>אנחנו לא עושים שום טעויו</del>ת. כאשר אנו עושים, אנחנו יכולים לכתוב את זה סמנטית סימון כי בכל מקרה.
    </p>

    <p>
      כאשר אנו משתמשים בקיצורי דרך מסוימים כמו <abbr title='נראה לי טוב'>LGTM</abbr>, אנו יכולים להציג מידע כזה עבור המשתמש.
    </p>

    <h2>ציטוטים ועדויות</h2>

    <blockquote>
      זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
    </blockquote>

    <blockquote>
      זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
      <footer>
        <cite>מה-שמו</cite>
      </footer>
    </blockquote>

    <div style={{ display: 'flex' }}>
      <blockquote style={{ marginLeft: '15px' }}>
        זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
  היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
  אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
        <footer>
          <cite>מה-שמו</cite>
        </footer>
      </blockquote>

      <blockquote>
        זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
  היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
  אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
        <footer>
          <cite>מה-שמו</cite>
        </footer>
      </blockquote>
    </div>
  </div>
))
