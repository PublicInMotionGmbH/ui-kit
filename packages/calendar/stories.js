import React from 'react'
import { createStoriesFactory, getReadmeDescription } from '@talixo/shared/story'

import Calendar from './src/Calendar'

// Load first paragraph from README file
const readme = getReadmeDescription(require('./README.md'))

// Create factories for story
const addStory = createStoriesFactory('Calendar', module)

// Styles for stories
const styles = {
  position: 'relative',
  display: 'inline-block',
  left: '1px',
  marginRight: '30px'
}

// Stories

addStory('default', readme, () => (
  <Calendar style={styles} />
))

addStory('custom date format', readme, () => (
  <div>
    <Calendar style={styles} displayFormat='YYYY/MM/D' />
    <Calendar style={styles} displayFormat='DD-MM-YYYY' />
    <Calendar style={styles} displayFormat='ddd MM YY' />
  </div>
))

addStory('custom placeholder', readme, () => (
  <Calendar style={styles} placeholder='Enter date' />
))
