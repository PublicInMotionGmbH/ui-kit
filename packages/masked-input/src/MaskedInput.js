import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

/**
 * Component which represents Masked Input.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @returns {React.Element}
 */
function MaskedInput (props) {
  const { className, ...passedProps } = props

  return (
    <span className={buildClassName('masked-input', className)} {...passedProps} />
  )
}

MaskedInput.propTypes = {
  /** Additional class name */
  className: PropTypes.string
}

MaskedInput.defaultProps = {
}

export default MaskedInput
