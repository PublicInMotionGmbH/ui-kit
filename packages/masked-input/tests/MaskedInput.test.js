import React from 'react'
import { shallow } from 'enzyme'

import { buildClassName } from '@talixo/shared'
import { Calendar } from '@talixo/calendar'

import MaskedInput, { moduleName } from '../src/MaskedInput'

const maskCls = buildClassName([moduleName, 'mask'])
const createProps = props => ({
  ...props
})
const createCalendar = (props = createProps()) => shallow(
  <MaskedInput {...props} renderMask={({ value }) => <div>{value.toString()}</div>}>
    <Calendar />
  </MaskedInput>
)

const createWrapper = (props, inputProps) => shallow(
  <MaskedInput {...props} renderMask={x => (<div className='test-mask'>{x.value}</div>)}>
    <input className='test-input' {...inputProps} />
  </MaskedInput>
)

describe('<MaskedInput />', () => {
  let wrapper
  it('renders children correctly', () => {
    const wrapper = createCalendar()
    expect(wrapper).toMatchSnapshot()
  })

  describe('state changes', () => {
    let input
    beforeEach(() => {
      wrapper = createCalendar()
      input = wrapper.find('Calendar')
    })

    it('should change state.focused to true', () => {
      input.simulate('focus')
      input.simulate('blur')
      input.simulate('focus')
      expect(wrapper.state().focused).toBe(true)
    })

    it('should change state.focused to true', () => {
      input.simulate('focus')
      input.simulate('blur')
      expect(wrapper.state().focused).toBe(false)
    })

    it('should change state.value', () => {
      const value = '2020-12-31'
      input.simulate('change', value)
      expect(wrapper.state().value).toBe(value)
    })
  })

  describe('masking behaviour when value prop is undefined', () => {
    let input

    beforeEach(() => {
      wrapper = createCalendar()
      input = wrapper.find('Calendar')
    })

    it('should apply mask when correct date is passed', () => {
      const value = '2020-12-12'
      input.simulate('change', value)
      expect(wrapper.find(`.${maskCls}`).exists()).toBe(true)
    })

    it('should not apply mask when no value is passed', () => {
      const valuePre = '2020-12-12'
      input.simulate('change', valuePre)
      const valuePost = ''
      input.simulate('change', valuePost)
      expect(wrapper.find(`.${maskCls}`).exists()).toBe(false)
    })
  })
  describe('masking behaviour on when value is passed', () => {
    beforeEach(() => {
      wrapper = createWrapper({ value: null })
    })

    it('should not display mask when null is default value', () => {
      expect(wrapper.find('.test-mask').exists()).toBe(false)
    })

    describe('when value prop is updated', () => {
      const value = 'Test value'
      beforeEach(() => {
        wrapper.setProps({ value })
      })

      it('should display mask with passed value', () => {
        expect(wrapper.find('.test-mask').exists()).toBe(true)
      })

      it('should update mask component value', () => {
        expect(wrapper.find('.test-mask').text()).toBe(value)
      })
      it('should remove mask when null is passed to value prop', () => {
        wrapper.setProps({ value: null })
        expect(wrapper.find('.test-mask').exists()).toBe(false)
      })
    })
  })

  describe('masked input event handling', () => {
    const onChange = jest.fn()
    const onBlur = jest.fn()
    const onFocus = jest.fn()
    let input

    beforeEach(() => {
      wrapper = createWrapper({ onChange, onBlur, onFocus })
      input = wrapper.find('.test-input')
    })
    it('should invoke children onChange', () => {
      input.simulate('change', { target: { value: 'test' } })
      expect(onChange).toHaveBeenCalledTimes(1)
    })
    it('should invoke children onFocus', () => {
      input.simulate('focus')
      expect(onFocus).toHaveBeenCalledTimes(1)
      expect(onFocus).toHaveBeenCalledWith(true)
    })
    it('should invoke children onBlur', () => {
      input.simulate('blur')
      expect(onBlur).toHaveBeenCalledTimes(1)
      expect(onBlur).toHaveBeenCalledWith(false)
    })
  })

  describe('children event handling', () => {
    const onChange = jest.fn()
    const onBlur = jest.fn()
    const onFocus = jest.fn()
    let input

    beforeEach(() => {
      wrapper = createWrapper({}, { onChange, onBlur, onFocus })
      input = wrapper.find('.test-input')
    })
    it('should invoke children onChange', () => {
      input.simulate('change', { target: { value: 'test' } })
      expect(onChange).toHaveBeenCalledTimes(1)
    })
    it('should invoke children onFocus', () => {
      input.simulate('focus')
      expect(onFocus).toHaveBeenCalledTimes(1)
    })
    it('should invoke children onBlur', () => {
      input.simulate('focus')
      input.simulate('change', { target: { value: 'test' } })
      input.simulate('blur')
      expect(onBlur).toHaveBeenCalledTimes(1)
    })
  })
})
