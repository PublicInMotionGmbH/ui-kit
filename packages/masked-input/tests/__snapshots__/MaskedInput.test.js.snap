// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaskedInput /> renders children correctly 1`] = `
<div
  className="talixo-masked-input"
>
  <Calendar
    displayFormat="YYYY-MM-DD"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
  />
</div>
`;
